lib/features/restauran_tour/model/restauran_service.dart

```diff
@@ -0,0 +1,78 @@
+import 'package:flutter/foundation.dart';
+import 'package:restaurantour/features/restauran_tour/model/restaurant.dart';
+import 'package:restaurantour/repositories/api_status.dart';
+import 'package:restaurantour/repositories/endpoints.dart';
+import 'package:restaurantour/repositories/yelp_repository.dart';
+
+class RestauranService {
+  late YelpRepository yelpRepo;
+
+  RestauranService({
+    @visibleForTesting YelpRepository? yelpRepo,
+  }) : yelpRepo = yelpRepo ?? YelpRepository();
+
+  Future<Object> getAllRestaurans({int offSet = 0}) async {
+    try {
+      var response = await yelpRepo.getResponse(
+        path: restauranUrl,
+        query: _getQuery(offSet),
+      );
+      if (200 == response.statusCode) {
```

Comment:
I know it's the same but it's a personal preference
```suggestion
      if (HttpStatus.ok == response.statusCode) {
```